flink-end-to-end-tests/flink-end-to-end-tests-sql/pom.xml

```diff
@@ -199,4 +199,29 @@
             </plugin>
         </plugins>
     </build>
+
+    <profiles>
+        <!-- Activate these profiles with -Phadoop3-tests to build and test against hadoop 3.1.3 -->
+        <profile>
+            <id>hadoop3-tests</id>
+            <properties>
+                <flink.hadoop.version>3.1.3</flink.hadoop.version>
+            </properties>
```

Comment:
you shouldn't set the property in the profile; it will leak to inconsistent behavior when we use a different Hadoop 3 version on CI.
```suggestion
```